### ISSUE TITLE
[UX] Introduce button group UI component

### ISSUE THREAD

**Issue by Al-Rozhkov** [backdrop/backdrop-issues#2643]
Backdrop community pay a lot of attention to UX issues. And it is priceless. As a result Backdrop is several steps ahead of Drupal (and many other CMS) in terms of UX.

I would like to propose new UI button group component for Backdrop. It is select component which act like radio buttons or checkboxes. You could see it many times before all over the internet. It is present in bootstrap css/js framework.

![bootstrap-button-groups](https://cloud.githubusercontent.com/assets/16973264/25066561/40fec5ce-223a-11e7-8e36-5e16c0242adf.jpg)

It is perfect solution when you have just few options to choose. Its main advantage that you can see all options immediately without additional clicking. Button group needs one click less than selectbox to provide action.

Here's how it might look on Manage content page. Both options are chosen. It is analog of "- Any -" option.

![manage-content-any](https://cloud.githubusercontent.com/assets/16973264/25066593/61909b68-223b-11e7-8096-9e368a668248.jpg)

Here's one option chosen.

![manage-content-published](https://cloud.githubusercontent.com/assets/16973264/25066594/7f2959bc-223b-11e7-9716-870a6d8d6266.jpg)

Manage user accounts page.

![manage-user-accounts](https://cloud.githubusercontent.com/assets/16973264/25066598/9b91533e-223b-11e7-989d-e1297d8c53dc.jpg)

This component can be used widely. It is just a sketch. Styles needs to be discussed.

**Comment by Al-Rozhkov** [comment]
With button group you can not unselect all options. It is perfectly fits to new design of modules page introduced in #37 by @docwilmot.

When you have one option selected it is disabled. Because there is no sence to have both options unselected.

![modules-page-one-option](https://cloud.githubusercontent.com/assets/16973264/25066675/36bfef44-223e-11e7-9928-c2df35fce604.jpg)

Here is how it might look with button group.

![modules-page-button-group](https://cloud.githubusercontent.com/assets/16973264/25066682/61e94698-223e-11e7-99e0-599ba6d7a7d4.jpg)

**Comment by klonos** [comment]
I do like the 1 less click. I also like the fact that as buttons, we have wider target area than a checkbox, so better support for thumb selection on mobile. It fits better with the "material" design that's in trend nowadays with android taking over the market.

The only thing that makes me skeptical is that these now look like primary buttons, taking focus out of the "Filter" button. But perhaps with some work on the styling we can fix this.

Any other downsides anyone else can think of?

**Comment by Al-Rozhkov** [comment]
> these now look like primary buttons

@klonos I agree. What about gray?

![manage-content-gray-buttons](https://cloud.githubusercontent.com/assets/16973264/25067331/0052dce2-2251-11e7-9cef-6e73952c88f7.jpg)

**Comment by klonos** [comment]
Yep, that's better and having the same shade of gray border as the other inputs makes it more obvious that these are in fact controls and not buttons. Would still like to see what others say though.

**Comment by olafgrabienski** [comment]
I like the idea! Would be nice to have a PR to get an impression of the look and feel in relation to other elements. For instance the *Manage Content* page with the three different filter types in a row.

**Comment by laryn** [comment]
I like it, too!
